src/csharp/Grpc.Core/Internal/NativeExtension.cs

```diff
@@ -18,165 +18,153 @@
+    /// Gets singleton instance of this class.
+    /// The native extension is loaded when called for the first time.
+    /// </summary>
+    public static NativeExtension Get() {
+      if (instance == null) {
+        lock (staticLock) {
+          if (instance == null) {
+            instance = new NativeExtension();
+          }
         }
+      }
+      return instance;
+    }
 
-        /// <summary>
-        /// Provides access to the exported native methods.
-        /// </summary>
-        public NativeMethods NativeMethods
-        {
-            get { return this.nativeMethods; }
-        }
+    /// <summary>
+    /// Provides access to the exported native methods.
+    /// </summary>
+    public NativeMethods NativeMethods {
+      get { return this.nativeMethods; }
+    }
 
-        /// <summary>
-        /// Detects which configuration of native extension to load and load it.
-        /// </summary>
-        private static UnmanagedLibrary Load()
-        {
-            // TODO: allow customizing path to native extension (possibly through exposing a GrpcEnvironment property).
-            // See https://github.com/grpc/grpc/pull/7303 for one option.
-            var assemblyDirectory = Path.GetDirectoryName(GetAssemblyPath());
-
-            // With old-style VS projects, the native libraries get copied using a .targets rule to the build output folder
-            // alongside the compiled assembly.
-            // With dotnet cli projects targeting net45 framework, the native libraries (just the required ones)
-            // are similarly copied to the built output folder, through the magic of Microsoft.NETCore.Platforms.
-            var classicPath = Path.Combine(assemblyDirectory, GetNativeLibraryFilename());
-
-            // With dotnet cli project targeting netcoreapp1.0, projects will use Grpc.Core assembly directly in the location where it got restored
-            // by nuget. We locate the native libraries based on known structure of Grpc.Core nuget package.
-            // When "dotnet publish" is used, the runtimes directory is copied next to the published assemblies.
-            string runtimesDirectory = string.Format("runtimes/{0}/native", GetPlatformString());
-            var netCorePublishedAppStylePath = Path.Combine(assemblyDirectory, runtimesDirectory, GetNativeLibraryFilename());
-            var netCoreAppStylePath = Path.Combine(assemblyDirectory, "../..", runtimesDirectory, GetNativeLibraryFilename());
-
-            // Look for all native library in all possible locations in given order.
-            string[] paths = new[] { classicPath, netCorePublishedAppStylePath, netCoreAppStylePath};
-            return new UnmanagedLibrary(paths);
-        }
+    /// <summary>
+    /// Detects which configuration of native extension to load and load it.
+    /// </summary>
+    private static UnmanagedLibrary Load() {
+      if (LibBridge.type != LibBridgeType.UnmanagedLibrary) {
+        return null;
```

Comment:
Apologies; my linter rather over-zealously edited this file. This should be the only line which is actually necessary.